### PR TITLE
implement nextcloud cached subscriptions

### PR DESCRIPTION
**Pull request by georgehrke**
Signed-off-by: Georg Ehrke <developer@georgehrke.com>

### CODE REVIEW

File: src/request.js [hunk 1]
```diff
@@ -525,6 +525,7 @@ function getDefaultHeaders() {
 
 	return {
 		Depth: '0',
-		'Content-Type': 'application/xml; charset=utf-8'
+		'Content-Type': 'application/xml; charset=utf-8',
+		'X-NC-CalDAV-Webcal-Caching': 'On'
```
Comment (tcitworld) [inline src/request.js]: Can't we set it only for calendar related requests and not as default ?

Comment (georgehrke) [reply src/request.js]: We could, but that would require much overriding plenty of `DavCollection` methods in Calendar. And sending it all the time should not do any harm for now :)

Comment (georgehrke) [reply src/request.js]: @tcitworld Are you fine with this change or should we change it? :)

Comment (tcitworld) [reply src/request.js]: Yup, we'll handle this later.

**Pull request merged**